### ISSUE TITLE
feature request: --print-filesystems 

### ISSUE THREAD

**Issue by chesty** [jimsalterjrs/sanoid#297]
Hi,

If there was a feature than did nothing but printed all the filesystems in sanod.conf, I think we could do something like 
```
syncoid `sanoid --print-filesystems` user@hosts:backup
```
there would be caveats like recursive = yes and autosnap = no, so no worried if it's not deemed useful or possible.

**Comment by crashbuggy** [comment]
fair enough, ```for source in `sanoid --print-filesystems`; do syncoid "$source" ..; done```

If I'm going to do that I may as well use awk to parse the config file.

atm I just sync the pool recursively, but I get critical errors on empty canmount=no and temporary filesystems that I exclude from sanoid snapshots, which don't cause any issues, but I'd rather a successful sync finish with 0 critical errors so it's easier to catch syncs that failed.

I'm still learning and experimenting on how best to check the backups are running ok and test they make it to the disk on the backup pool. I'm of the school of thought that says you don't have backups until you test them.

cheers.

**Comment by chesty** [comment]
This time with my usual account.

fair enough, for source in `sanoid --print-filesystems`; do syncoid "$source" ..; done

If I'm going to do that I may as well use awk to parse the config file.

atm I just sync the pool recursively, but I get critical errors on empty canmount=no and temporary filesystems that I exclude from sanoid snapshots, which don't cause any issues, but I'd rather a successful sync finish with 0 critical errors so it's easier to catch syncs that failed.

I'm still learning and experimenting on how best to check the backups are running ok and test they make it to the disk on the backup pool. I'm of the school of thought that says you don't have backups until you test them.

cheers.

**Comment by phreaker0** [comment]
@chesty 
you can set a property on the datasets you don't want to be replicated:
`zfs set syncoid:sync=false pool/dataset`
**NOTE**: the property will be inherited, so you have to unset/set them on the child datasets as needed as well, or you will ignore to much

**Comment by chesty** [comment]
sweet, thanks. that'll work.
I can turn on sync where I want including /var/lib/docker which will turn on sync by inheritance on /var/lib/docker/0fe4a28..temporary.randomly.named.dataset, then exclude them with --exclude=rpool/var/lib/docker/[a-f0-9]{64} --exclude=rpool/var/lib/docker/[a-f0-9]{64}-init

**Comment by phreaker0** [comment]
if you use the exclude arguments you won't really need the sync property stuff, I didn't think of the exclude argument, despite having implemented it :-D

**Comment by chesty** [comment]
I think I tested using exclude, it didn't do what I wanted, for example I don't want rpool/var sync'd, but I do want rpool/var/lib/docker, from memory if I added --exclude=rpool/var it excludes rpool/var/lib/docker too.
I just thought maybe --exclude=rpool/var$ would work? I'm not sure, I haven't tested, I'm happy with using the syncoid:sync property anyway.
thanks, one step closer to mastering syncoid and sanoid.
